src/htm/algorithms/Connections.hpp

```diff
@@ -510,7 +510,7 @@ class Connections : public Serializable
    */
   void synapseCompetition(  const Segment    segment,
                             const SynapseIdx minimumSynapses,
-                            const SynapseIdx maximumSynapses);
+                            const SynapseIdx maximumSynapses); 
```

Comment:
TODO 
- [ ] this should have some parametric default values
- [ ] replace (or provide overload) with percentages off the range [stimulusThreshold, maxSynapsesPerSegment]?